dd-library-php-setup.php

```diff
@@ -189,22 +227,147 @@ function install($options)
             echo "Installation to '$binaryForLog' was successful\n";
         }
     }
+}
 
-    echo "--------------------------------------------------\n";
-    echo "SUCCESS\n\n";
-    if ($interactive) {
-        echo "Run this script in a non interactive mode adding the following 'php-bin' options:\n";
-        $args = array_merge(
-            $_SERVER["argv"],
-            array_map(
-                function ($el) {
-                    return '--php-bin=' . $el;
-                },
-                array_keys($selectedBinaries)
-            )
+function install_appsec($options, $selectedBinaries)
+{
+    if ($options[OPT_NO_APPSEC]) {
+        return;
+    }
+    if (
+        !isset($options[OPT_APPSEC_FILE]) &&
+        !isset($options[OPT_APPSEC_URL]) &&
+        !isset($options[OPT_APPSEC_VERSION]) ||
+        isset($options[OPT_APPSEC_VERSION]) && $options[OPT_APPSEC_VERSION] == 'latest'
+    ) {
+        $options[OPT_APPSEC_VERSION] = latest_release("dd-appsec-php");
+    }
+
+    echo "Installing appsec\n";
+
+    // Preparing clean tmp folder to extract files
+    $tmpDir = sys_get_temp_dir() . '/dd-appsec';
+    $tarball = "$tmpDir/dd-appsec-php.tar.gz";
+    execute_or_exit(
+        "Cannot create directory '$tmpDir'",
+        "mkdir -p " . escapeshellarg($tmpDir)
+    );
+    execute_or_exit(
+        "Cannot clean '$tmpDir'",
+        "rm -f " . escapeshellarg($tarball)
+    );
+    delete_on_exit($tmpDir);
+
+    // Retrieve the archive to the temporary location
+    if (isset($options[OPT_APPSEC_FILE])) {
+        $tarball = $options[OPT_APPSEC_FILE];
+    } else {
+        $url = isset($options[OPT_APPSEC_URL])
+            ? $options[OPT_APPSEC_URL]
+            : sprintf(
+                'https://github.com/DataDog/dd-appsec-php/releases/' .
+                "download/v%s/dd-appsec-php-%s-amd64.tar.gz",
+                rawurlencode($options[OPT_APPSEC_VERSION]),
+                rawurlencode($options[OPT_APPSEC_VERSION])
```

Comment:
I don't like redundant sprintf args :-)
```suggestion
                'https://github.com/DataDog/dd-appsec-php/releases/download/v%1$s/dd-appsec-php-%1$s-amd64.tar.gz',
                rawurlencode($options[OPT_APPSEC_VERSION])
```